### DISCUSSION TITLE
Depedency Track to monitor Wordpress Plugins

### DISCUSSION THREAD

**Discussion by miftacul** [DependencyTrack/dependency-track#1600] [General]
Hi All,

Is there any roadmap or improvement plan for DT able to monitor Wordpress Plugins Vulnerability?

![image](https://user-images.githubusercontent.com/17409741/167593764-0f0e242f-82ff-4090-8370-6777bb2aeef4.png)

I find that there is tool called [WpBom ](https://wordpress.org/plugins/wpbom/) in the CycloneDX tool center that able generate SBOM for wordpress plugins and also integrate with DT as well. 

![image](https://user-images.githubusercontent.com/17409741/167594158-1a491a96-530f-4df5-8029-e14451f6694d.png)
I have installed and implement it successfully. But i'm not sure whether it can help us to monitor the wordpress plugin vulnerability or not.

or, Is DT really already able to monitor Wordpress Plugin Vulnerability?

Thank You

**Comment by stevespringett** [comment]
DT is agnostic to the language or ecosystem. If that wordpress plugin generates proper inventory for installed plugins and that inventory includes the correct package url or CPE, then DT will be able to identify vulnerabilities in them. If this doesn't work for you, you may want to contact the author.

**Reply by gschafra** [reply]
PURLs are generated like this: `pkg:wordpress/plugins/contact-form-7@5.1.4`. Currently no CPE is generated. Unfortunately vulnerabilities existing on this plugin with this version (see https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-35489) are not detected when only PURL is given. When manually supplement the CPE (`cpe:2.3:a:rocklobster:contact_form_7:5.1.4:*:*:*:*:wordpress:*:*`), the vulnerability is instantly shown/detected. So... unfortunately WP-plugins do not provide vendor names, so it is somewhat awkward to generate a valid CPE. But this should be an issue of the [wpbom project](https://github.com/sepbit/wpbom).

**Reply by stevespringett** [reply]
The purl type `wordpress` is not currently defined, but has been reserved.
https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst

This means that there's a high probability that no source of vulnerability intelligence is using the wordpress PURL type to associate components with vulnerabilities. This will not happen overnight. The PURL type needs to be property defined. Sources of vuln intel need to adopt it and all the data science behind it.

It might be easier to modify the wpbom project to include simple CPE dictionary mappings.

`contact-form-7` would map to `rocklobster:contact_form_7`.

**Reply by gschafra** [reply]
... as suggested in https://github.com/sepbit/wpbom/issues/1. Thanks for the purl-spec info and your advise.

**Comment by gschafra** [comment]
Just opened https://github.com/sepbit/wpbom/issues/1

**Comment by vitoranguia** [comment]
Hello! How are you?
Thanks for the alert!
I replied about the problem in sepbit/wpbom#1
